pkg/config/lvmd/lvmd.go

```diff
@@ -0,0 +1,104 @@
+// DeviceClass type definition and validation is taken from:
+// https://github.com/red-hat-storage/topolvm/blob/release-4.10/lvmd/device_class_manager.go
+package lvmd
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultSpareGB  = 10
+	defaultSockName = "/run/lvmd/lvmd.socket"
+)
+
+var qualifiedNameRegexp = regexp.MustCompile("^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$")
+
+// This regexp is used to check StripeSize format
+var stripeSizeRegexp = regexp.MustCompile("(?i)^([0-9]*)(k|m|g|t|p|e|b|s)?$")
+
+// DeviceClass maps between device-classes and volume groups.
+type DeviceClass struct {
+	// Name for the device-class name
+	Name string `yaml:"name"`
+	// Volume group name for the device-class
+	VolumeGroup string `yaml:"volume-group"`
+	// Default is a flag to indicate whether the device-class is the default
+	Default bool `yaml:"default"`
+	// SpareGB is storage capacity in GiB to be spared
+	SpareGB uint `yaml:"spare-gb"`
```

Comment:
Should this be a pointer so we can tell the difference between being willing to not leave any spare disk and not providing a value at all? Or does the value have to be > 0 to be valid?

Reply:
Lvmd will default a null `spare-gb` field to 10Gb, so users will have explicitly define the spare-gb field for any other value.

Reply:
How does that work? With the struct the way it is, I would expect the value of `SpareGB` to be `0` if the user does not provide anything. Then when we render the struct, I would expect to see the explicit `0` passed it to lvmd. Does lvmd convert the 0 to 10?

Reply:
In lvmd, non-nil, non-negative values are valid inputs, and a nil value results in a 10gb default being used. IMO that seems overly cautious.  In prod, I would expect the system storage to remain static after the device is imaged, with the unallocated VG space being managed by lvmd.  Meaning the spared space would never be utilized. If that's a valid assumption, then defaulting to 0 seems reasonable.